woudc_data_registry/models.py

```diff
@@ -1163,12 +1250,113 @@ def sync(ctx):
+        contributor_dict = {contributor_models[i].contributor_id:
+                            contributor_models[i]
+                            for i in range(0, len(contributor_models))}
+
+        """deployment_dict will have stations as
+        keys in order to help connect them to projects """
+        deployment_dict = {deployment_models[i].station_id:
+                           deployment_models[i]
+                           for i in range(0, len(deployment_models))}
+
+        """contribution dict used to check for duplicate contribution id"""
+        contribution_dict = {contribution_models[i].contribution_id:
+                             contribution_models[i]
+                             for i in range(0, len(contribution_models))}
+
+        for key in instrument_dict:
+
+            for station in station_models:
+
+                """station info"""
+                station_id = station.station_id
+                station_name = station.station_name_id
+
+                """country info"""
+                country_id = station.country_id
+                country_name_en = country_dict[country_id].name_en
+                country_name_fr = country_dict[country_id].name_fr
+
+                """instrument info"""
+                instrument_id = instrument_dict[key].instrument_id
+                instrument_name = instrument_dict[key].name
+                instrument_model = instrument_dict[key].model
+
+                start_date = instrument_dict[key].start_date
+                end_date = instrument_dict[key].end_date
+                dataset_id = instrument_dict[key].dataset_id
+
+                x = instrument_dict[key].x
+                y = instrument_dict[key].y
+                z = instrument_dict[key].z
+
+                """ get the contributor id first from
+                the deployment at the specific station"""
+                contributor_id = \
+                    deployment_dict[station.station_id].contributor_id
+
+                """ now access the project from contributor"""
+                project_id = contributor_dict[contributor_id].project_id
+
+                """form the contribution id by combining the
+                strings present in Contributions dependencies"""
+                contribution_id = project_id + ":" + dataset_id + ":" \
+                    + station_id + ":" + instrument_name
+
+                """check if the contribution id is in the index already"""
+                if contribution_id in contribution_dict.keys():
+
+                    """if it is then update the start
+                    and end date for that contribution id"""
+                    for index in contribution_models:
+                        if contribution_models[index].contribution_id \
+                                == contribution_id:
+                            contribution_models[index].start_date = start_date
+                            contribution_models[index].end_date = end_date
```

Comment:
Can simply do `contribution_dict[contribution_id]` to get the right model right off the bat.

Also remember to only allow the `start_date` to decrease and the `end_date` to increase (`end_date` might be None as well), the way it is now changes both of them whether they increase or decrease.